Source/objects.cpp

```diff
@@ -321,6 +321,15 @@ bool RndLocOk(int xp, int yp)
 	return IsNoneOf(leveltype, DTYPE_CATHEDRAL, DTYPE_CRYPT) || dPiece[xp][yp] <= 125 || dPiece[xp][yp] >= 143;
 }
 
+bool IsAreaOk(Rectangle rect)
+{
+	for (Point position : PointsInRectangle(rect)) {
+		if (!RndLocOk(position.x, position.y))
+			return false;
+	}
+	return true;

```

Comment:
```suggestion
    const points = PointsInRectangle(rect);
    return std::any_of(points.begin(), points.end(), [](Point point) { return RndLocOk(point.x, point.y); });
```

(and `#include <algorithm>`)

Reply:
I've add `c_all_of` in #6450, which will make the following possible:

``c++
return c_all_of(PointsInRectangle(rect), &RndLocOk);
```

Reply:
Lets do that as a follow up?